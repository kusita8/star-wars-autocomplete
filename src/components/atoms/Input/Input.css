.input {
  width: 100%;
  padding: 0.6rem 0.8rem;
}
