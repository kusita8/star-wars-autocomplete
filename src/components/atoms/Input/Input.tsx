import { FC, ForwardedRef, forwardRef, InputHTMLAttributes } from "react";
import "./Input.css";

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
  label?: string;
  ref?: ForwardedRef<HTMLInputElement>;
}

export const Input: FC<InputProps> = forwardRef(({ label, name, ...props }, ref) => {
  return (
    <div>
      {label && <label htmlFor={name}>{label}</label>}
      <input ref={ref} className="input" type="text" id={name} name={name} {...props} />
    </div>
  );
});
