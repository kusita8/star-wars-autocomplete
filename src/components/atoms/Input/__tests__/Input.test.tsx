import { render } from "@testing-library/react";
import { Input } from "../Input";

describe("Input", () => {
  test("it should match the snapshot", () => {
    const { asFragment } = render(<Input label="hola mundo" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
