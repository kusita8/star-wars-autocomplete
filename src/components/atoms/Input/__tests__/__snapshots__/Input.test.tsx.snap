// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input it should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <label>
      hola mundo
    </label>
    <input
      class="input"
      type="text"
    />
  </div>
</DocumentFragment>
`;
